tests/build-test-report.js

```diff
@@ -254,6 +248,68 @@ function runTests(outputFilePath, verboseOutputFilePath) {
   });
 }
 
+function downloadCircleResults() {
+  console.log("downloading test results from circle ci...");
+  const VCS_TYPE = 'github';
+  const USERNAME = 'code-dot-org';
+  const PROJECT = 'JS-Interpreter';
+  const REQUEST_PATH = `https://circleci.com/api/v1.1/project/${VCS_TYPE}/${USERNAME}/${PROJECT}/${argv.circleBuild}/artifacts`;
+
+
+  return new Promise((resolve, reject) => request(
```

Comment:
You'd save yourself some funky promise wrapping with [`node-fetch`](https://www.npmjs.com/package/node-fetch).